### PR TITLE
Stuff

### PR DESCRIPTION
**Pull request by agilob**
So I did a lot of refactoring in many places of the application.

List of changes:
* removed 'self-copyright' statements which have no power - @SecUpwN  as project have all IP rights, this is GNUGPL project so we can't just write '2015 copyright agilob' otherwise me (agilob) or @SecUpwN are breaking the licence.
*  code refactoring in smsdetector (package), more information from exceptions, removed unused code, no changes in how it works
* moved background thread from 'inline' implementation to separate inner class extending ASyncTask, so onCreate is shorted and whole OpenCellIdActivity looks better
* removed more unused code and imports
* see commits and changes

### CODE REVIEW

**Comment by agilob** [comment]
omg so many conflicts :<
I'll solve it different way.

**Comment by E3V3A** [comment]
It's a very bad idea to remove code when work are in progress, especially when made by another developer. Please ask original developer before removing this kind of code. There are plenty of open and far more important issues to handle than this! 

@banjaxbanjo Are you ok with this?

**Comment by E3V3A** [comment]
@SecUpwN  Please do not merge this PR!! This is BS as it removes tons of comments very much needed and messes up the formatting!

**Comment by agilob** [comment]
That's why I put it in another commit.

**Comment by SecUpwN** [comment]
>This is BS as it removes tons of comments very much needed and messes up the formatting!

@agilob, please don't take this "rant" too personal, all he means is that we should not remove code and code comments of other developers if they have not been asked in the PR. Thanks for continuing to work on refactoring and cleaning our code, please add proper code comments and make sure to **name** your PR properly as well - "Stuff" is not a good thing to find later on if it needs to be tracked or referenced. ;-)

**Comment by E3V3A** [comment]
@agilob Yes, sorry, don't take it personal. We're just trying to maintain some kind of standard here, both regarding the code and the comments within it. There's lot's of WIP stuff in our code and we've had too many problems with poorly documented code in the past. I've spent considerable time trying to document as much as I could and that is still far from satisfactory.

**Comment by banjaxbanjo** [comment]
lol I like to put my stamp on work I've done and it doesn't mean that the code is protected even tho I put in copyright etc.. What was the point of taking of my name from the work I've worked hard to create? It also is helpful rather than having to email and find who created what class etc.. I find it an insult to remove my name from the classes I created especially app/src/main/java/com/SecUpwN/AIMSICD/smsdetection/SmsDetector.java with is a major class that does detect sms.

**Comment by agilob** [comment]
No other classes have name on the top of the file so I removed it to keep everything consistent, don't take it as an insult. You claimed your copyright there, which makes your code is breaking GPL licence of the whole project (btw. that's not the only OS licence that is broke in this project...). If you want to keep your name there, just add header of an open source licence under your name. That's how it works in every opensource project. Work belongs to the owner of the repository unless they use 3rd party code - in such case application must have 'credits' view, where your name have to be listed.

I thought you're not going to fix and extend your code. There is a lot of unused code (which I removed). Even after I asked you to improve that code - fix exceptions, formatting, removed ~10 unused variables, you didn't do it in >3 weeks.

To keep track of who works on which part of the application it's enough to `git blame app/src/main/java/com/SecUpwN/AIMSICD/smsdetection/SmsDetector.java` and all authors of every line will be printed with dates of last modification.

To sum up, your code is not finished, is bloated, you're confusing System.out with Log.e/Log.i/Log.d, you're not printing exceptions and error messages carefully in such critical code. You didn't touch that code since it was merged and it's breaking GPL licence. Don't take it as an insult, I had time today and decided to fix all of those issues. If anyone wants to fix it in the future, feel free to use my commits.

**Comment by banjaxbanjo** [comment]
On Holidays mate and this is not my full time job we all pitch in here and there sometimes we cant fix stuff asap (RL gets in way kids etc..). This app is WIP also and in not finished and my code for detection is definitely not finished as I've said many times its only a base class to give us something to work off. I did also push wap push detection so I don't know about not extending code..

Does writing my name without the copyright text in a class break the GPL license? if not what harm is giving myself a little credit and what harm is a little gray comment string going to do:) .

**Comment by E3V3A** [comment]
> To keep track of who works on which part of the application it's enough to git blame...

@agilob Unfortunately that doesn't always work as the code has been "refactored" (I hate that word) several times, and all history has been lost or jumbled up.

A simple `change-log` should be present in all code files, including a description of it's use, functionality and current status. If it says "todo", then it stays until original developer removes it. [This](https://github.com/SecUpwN/Android-IMSI-Catcher-Detector/blob/development/app/src/main/java/com/SecUpwN/AIMSICD/fragments/AtCommandFragment.java#L64-L67) is how we've done it before: For example:

```
 *  ChangeLog:
 *              2015-02-11  E:V:A       Testing to add back some old working code
 *              2015-02-16  scintill    Made some code changes and added /utils/atcmd/...
 *              2015-02-11  E:V:A       Added 10 minutes timeout
```

Furthermore its a good idea to leave emails out from code block as you **will** get spammed. Put your name/tags whatever you call it, but try to leave emails out. 

I strongly suggest to use our HipChat or other direct communication means, for developer talks for more clear communication before pushing PRs that involve a lot of changes to code by other active developers.

**Comment by SecUpwN** [comment]
>To sum up, your code is not finished, is bloated, you're confusing System.out with Log.e/Log.i/Log.d, you're not printing exceptions and error messages carefully in such critical code. You didn't touch that code since it was merged and it's breaking GPL licence. Don't take it as an insult, I had time today and decided to fix all of those issues. If anyone wants to fix it in the future, feel free to use my commits.

@agilob, thanks for pointing out the difficulties in our current code. I owe you very much *respect* that you did so much digging and I know that you may be a little sick of how the code looks in general. I strongly encourage you to wait until we have finished the discussion on how to proceed here as well as in #476, then feel invited to re-craft changes in preferably smaller pull requests that are a little easier to check for us and do not conflict with the current changes. It would be very nice of you if you get in touch with @banjaxbanjo viia email and clear up what he shall fix so that both of you step ahead to resolve the hurdles between you. Hint: Always *pull* the latest changes with my beloved [SmartGit](http://www.syntevo.com/smartgit/) onto your local storage *before* changing the files and then craft a new PR with a *push* - this ensures consistency. Please reference this closed PR in your new one so that we can track this. Again, thank you very much for your hard work!

>please email me personally if you have a problem with something I need to fix I don't scan the comments as much as I should and a direct email is more affective. We are all here to work together because we have the same interests, a little heads up for me by PM would be nice as I am a novice programmer if something is not right.

@banjaxbanjo, perfectly true point. This is an open project and all developers not only have a private life, but also their own moods. So whatever we do here and whatever we discover that is off, please handle each other carefully (and I literally expect that from *everybody*). Yes, a private email is much better to handle (harder) criticism, that's what we have our [contact page](https://github.com/SecUpwN/Android-IMSI-Catcher-Detector/wiki/Contact) for. Should an email not be listed there, get in touch with me. Also, I'm willing to chime in as a "mediator" between you and someone else at any time: Just get in touch with me via private email and I'll do *everything* to resolve any misunderstanding or hurt feelings. Not telling me how you feel is *not* a solution. Feelings are important.

>Furthermore its a good idea to leave emails out from code block as you will get spammed. Put your name/tags whatever you call it, but try to leave emails out. 

@E3V3A, I thought it might be good having a line like this is much better to get in touch later on?

```
Coded by Mr. Android <example@android.com>
```

Shall I remove the emails from our code and rather add the link to the devs GitHub account like so?

```
Coded by Mr. Android <https://github.com/Android>
```

**Comment by E3V3A** [comment]
Well, I'm thinking that since we have [this](https://github.com/SecUpwN/Android-IMSI-Catcher-Detector/wiki/Contact), there's little point to also add it in the code. People can look up the github name there to get the email, if needed, and also easily change it to some anonymous (like opayq) email as well. 

Spam scripts are used for scraping Github repos for email accounts...so you do what you wish...

**Comment by banjaxbanjo** [comment]
Guys I know my code is like an alien language and hard to understand but feel free to contact me anytime about any particular line and I will tell you what its doing, I try not to over comment the code to show what every 5 lines is doing but I should be more clearer on hard to understand blocks.(WIP)

I'm also all for changes to my classes especially smsdetection class, if a PR fixes something , lightens up code or does it a better way and doesn't affect detection I'm all for that. The more stream line the code the better.
